request/request.go

```diff
@@ -128,3 +128,18 @@ func GetExtraResources(req *v1beta1.RunFunctionRequest) (map[string][]resource.E
 	}
 	return out, nil
 }
+
+// GetCredential from the supplied request.
+func GetCredential(req *v1beta1.RunFunctionRequest, name string) (resource.Credential, error) {
```

Comment:
I think this should be `GetCredentials` and return `resource.Credentials`.

Reply:
Yeah, I think the API was a bit confusing to me. You get a credential, which can have a map of credentials in it. I agree on using the `s` since it returns a map, but intuitively I'd think "if I filter on a name, I get one `credential` back".